addons/spreadsheet/static/tests/global_filters/global_filters_model_test.js

```diff
@@ -2060,6 +2060,37 @@ QUnit.module("spreadsheet > Global filters model", {}, () => {
         }
     );
 
+    QUnit.test("getFiltersMatchingPivot works with date=false", async function (assert) {
+        patchDate(2022, 6, 14, 0, 0, 0);
+        const { model } = await createSpreadsheetWithPivot({
+            arch: /*xml*/ `
+                <pivot>
+                    <field name="product_id" type="row"/>
+                    <field name="probability" type="measure"/>
+                    <field name="empty_date" interval="month" type="col"/>
+                </pivot>`,
+        });
+
+        await addGlobalFilter(
+            model,
+            {
+                id: "43",
+                type: "date",
+                label: "date filter 1",
+                rangeType: "fixedPeriod",
+                defaultValue: "this_month",
+            },
+            {
+                pivot: { 1: { chain: "empty_date", type: "date" } },
+            }
+        );
+        const dateFilters1 = getFiltersMatchingPivot(
+            model,
+            '=ODOO.PIVOT.HEADER(1,"expected_revenue","empty_date:month","false")'
```

Comment:
you actually don't need a new field for which there's no values. `=ODOO.PIVOT.HEADER(1,"expected_revenue","date:month","false")` is always a valid formula.
```suggestion
                    <field name="date" interval="month" type="col"/>
                </pivot>`,
        });

        await addGlobalFilter(
            model,
            {
                id: "43",
                type: "date",
                label: "date filter 1",
                rangeType: "fixedPeriod",
                defaultValue: "this_month",
            },
            {
                pivot: { 1: { chain: "date", type: "date" } },
            }
        );
        const dateFilters1 = getFiltersMatchingPivot(
            model,
            '=ODOO.PIVOT.HEADER(1,"expected_revenue","date:month","false")'
```